firestore/swift/firestore-smoketest/SolutionGeoPointViewController.swift

```diff
@@ -23,83 +23,80 @@ import CoreLocation
+  }
 
-    func geoQuery() {
-        // [START fs_geo_query_hashes]
-        // Find cities within 50km of London
-        let center = CLLocationCoordinate2D(latitude: 51.5074, longitude: 0.1278)
-        let radiusInM: Double = 50 * 1000
-
-        // Each item in 'bounds' represents a startAt/endAt pair. We have to issue
-        // a separate query for each pair. There can be up to 9 pairs of bounds
-        // depending on overlap, but in most cases there are 4.
-        let queryBounds = GFUtils.queryBounds(forLocation: center,
-                                              withRadius: radiusInM)
-        let queries = queryBounds.map { bound -> Query in
-            return db.collection("cities")
-                .order(by: "geohash")
-                .start(at: [bound.startValue])
-                .end(at: [bound.endValue])
-        }
+  func geoQuery() async {
+    // [START fs_geo_query_hashes]
+    // Find cities within 50km of London
+    let center = CLLocationCoordinate2D(latitude: 51.5074, longitude: 0.1278)
+    let radiusInM: Double = 50 * 1000
 
-        var matchingDocs = [QueryDocumentSnapshot]()
-        // Collect all the query results together into a single list
-        func getDocumentsCompletion(snapshot: QuerySnapshot?, error: Error?) -> () {
-            guard let documents = snapshot?.documents else {
-                print("Unable to fetch snapshot data. \(String(describing: error))")
-                return
-            }
+    // Each item in 'bounds' represents a startAt/endAt pair. We have to issue
+    // a separate query for each pair. There can be up to 9 pairs of bounds
+    // depending on overlap, but in most cases there are 4.
+    let queryBounds = GFUtils.queryBounds(forLocation: center,
+                                          withRadius: radiusInM)
+    let queries = queryBounds.map { bound -> Query in
+      return db.collection("cities")
+        .order(by: "geohash")
+        .start(at: [bound.startValue])
+        .end(at: [bound.endValue])
+    }
 
-            for document in documents {
-                let lat = document.data()["lat"] as? Double ?? 0
-                let lng = document.data()["lng"] as? Double ?? 0
-                let coordinates = CLLocation(latitude: lat, longitude: lng)
-                let centerPoint = CLLocation(latitude: center.latitude, longitude: center.longitude)
+    var matchingDocs = [QueryDocumentSnapshot]()
 
-                // We have to filter out a few false positives due to GeoHash accuracy, but
-                // most will match
-                let distance = GFUtils.distance(from: centerPoint, to: coordinates)
-                if distance <= radiusInM {
-                    matchingDocs.append(document)
-                }
-            }
-        }
+    // After all callbacks have executed, matchingDocs contains the result. Note that this code
+    // executes all queries serially, which may not be optimal for performance.
```

Comment:
Did you consider using a task group? 

https://www.hackingwithswift.com/quick-start/concurrency/how-to-create-a-task-group-and-add-tasks-to-it

Reply:
Done.